src/os/FuseStore.cc

```diff
@@ -594,59 +623,77 @@ static int os_open(const char *path, struct fuse_file_info *fi)
 
   case FN_OBJECT_DATA:
     {
-      pbl = new bufferlist;
-      int r = fs->store->read(cid, oid, 0, 0, *pbl);
+      int r = fs->open_file(
+	path, fi,
+	[&](bufferlist *pbl) {
```

Comment:
open_file fills in fi->fh.. so that ending block isn't used/needed for the object_data case